packages/in_app_purchase/ios/Classes/InAppPurchasePlugin.m

```diff
@@ -70,11 +123,150 @@ - (void)handleProductRequestMethodCall:(FlutterMethodCall *)call result:(Flutter
                                  details:call.arguments]);
       return;
     }
+    for (SKProduct *product in response.products) {
+      [self.productsCache setObject:product forKey:product.productIdentifier];
+    }
     result([FIAObjectTranslator getMapFromSKProductsResponse:response]);
     [weakSelf.requestHandlers removeObject:handler];
   }];
 }
 
+- (void)createPaymentWithProductID:(FlutterMethodCall *)call result:(FlutterResult)result {
+  if (![call.arguments isKindOfClass:[NSString class]]) {
+    result([FlutterError
+        errorWithCode:@"storekit_invalide_argument"
+              message:@"Argument type of createPaymentWithProductID is not a string."
+              details:call.arguments]);
+    return;
+  }
+  NSString *productID = call.arguments;
+  SKProduct *product = [self.productsCache objectForKey:productID];
+  if (!product) {
+    result([FlutterError
+        errorWithCode:@"storekit_product_not_found"
+              message:@"Cannot find the product. To create a payment of a product, you must query "
+                      @"the product with SKProductRequestMaker.startProductRequest."
```

Comment:
nit, I think this would be a little clearer.

```suggestion
                      @"the product with SKProductRequestMaker.startProductRequest first."
```